src/EditorFeatures/Core/FindUsages/AbstractFindUsagesService.cs

```diff
@@ -126,40 +126,55 @@ protected AbstractFindUsagesService(IThreadingContext threadingContext)
         /// and want to push all the references to it into the Streaming-Find-References window.
         /// </summary>
         public static async Task FindSymbolReferencesAsync(
-            IThreadingContext threadingContext,
-            IFindUsagesContext context, ISymbol symbol, Project project, CancellationToken cancellationToken)
+            IThreadingContext threadingContext, IFindUsagesContext context,
+            ISymbol symbol, Project project, CancellationToken cancellationToken)
         {
+            var monikerUsagesService = project.Solution.Workspace.Services.GetRequiredService<IFindSymbolMonikerUsagesService>();
+
             await context.SetSearchTitleAsync(string.Format(EditorFeaturesResources._0_references,
                 FindUsagesHelpers.GetDisplayName(symbol))).ConfigureAwait(false);
 
             var options = FindReferencesSearchOptions.GetFeatureOptionsForStartingSymbol(symbol);
-            var progressAdapter = new FindReferencesProgressAdapter(threadingContext, project.Solution, context, options);
+
+            // We're doing to kick off two find-references here.  One that roslyn performs, and one
+            // that goes through rich-nav.  In order for things like progress to work, we have each
+            // subsystem call into their context object.  These then get aggregated and passed along
+            // to the real context object.
+            var aggregator = new FindUsagesContextAggregator(context);
+            var roslynFindUsagesContext = aggregator.CreateForwardingContext();
+            var codeIndexFindUsagesContext = aggregator.CreateForwardingContext();
 
             // Now call into the underlying FAR engine to find reference.  The FAR
             // engine will push results into the 'progress' instance passed into it.
             // We'll take those results, massage them, and forward them along to the 
             // FindReferencesContext instance we were given.
-            await SymbolFinder.FindReferencesAsync(
+            var normalFindReferencesTask = SymbolFinder.FindReferencesAsync(
                 SymbolAndProjectId.Create(symbol, project.Id),
                 project.Solution,
-                progressAdapter,
+                new FindReferencesProgressAdapter(threadingContext, project.Solution, roslynFindUsagesContext, options),
                 documents: null,
-                options: options,
-                cancellationToken: cancellationToken).ConfigureAwait(false);
+                options,
+                cancellationToken);
+
+            // Kick off work to search the online code index system in parallel
+            var codeIndexReferencesTask = FindSymbolMonikerReferencesAsync(
+                monikerUsagesService,
+                symbol,
+                codeIndexFindUsagesContext,
+                cancellationToken);
+
+            await Task.WhenAll(normalFindReferencesTask, codeIndexReferencesTask).ConfigureAwait(false);
```

Comment:
i made a design decision here to just kick off a rich-nav search for the symbolmoniker the user started their search on.  So, specifically, rich-nav results are *not* cascading results.

We could change this around (if we thought it was important) and have us kick off rich-nav searches for *every* definition we cascade to as the main find-refs search progresses.  This would def be more complex implementation wise.  

For now, i think this is a acceptable starting point.

Reply:
Makes sense for now, or at least we can do this _after_ you've fleshed out the JSON bits.